src/core_codemods/combine_startswith_endswith.py

```diff
@@ -38,12 +43,19 @@ def matches_startswith_endswith_or_pattern(
     ) -> bool:
         # Match the pattern: x.startswith("...") or x.startswith("...")
         # and the same but with endswith
-        call = m.Call(
-            func=m.Attribute(
-                value=m.Name(), attr=m.Name("startswith") | m.Name("endswith")
-            ),
-            args=[m.Arg(value=m.SimpleString())],
+        args = [m.Arg(value=m.SimpleString())]
+        startswith = m.Call(
+            func=m.Attribute(value=m.Name(), attr=m.Name("startswith")),
+            args=args,
+        )
+        endswith = m.Call(
+            func=m.Attribute(value=m.Name(), attr=m.Name("endswith")),
+            args=args,
+        )
+        startswith_or = m.BooleanOperation(
+            left=startswith, operator=m.Or(), right=startswith
         )
-        return m.matches(
-            node, m.BooleanOperation(left=call, operator=m.Or(), right=call)
+        endswith_or = m.BooleanOperation(left=endswith, operator=m.Or(), right=endswith)
+        return m.matches(node, startswith_or | endswith_or) and m.matches(
+            node.left.func.value, node.right.func.value
```

Comment:
When using `matchers.matches` you should use the mock nodes from the `matchers` module as recommended by the docs:
https://libcst.readthedocs.io/en/latest/matchers.html#libcst.matchers.matches
Suggestion:
```python
node.left.func.value.value == node.right.func.value.value
```